### PR TITLE
tools/test_tpm_ek_chain.sh: script for checking EK cert chain size

### PR DESCRIPTION
**Pull request by krystian-hebel**

### CODE REVIEW

File: tools/test_tpm_ek_chain.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+else
+	# 3.0.2
+	pcrread="tpm2_pcrlist"
+	nvread="tpm2_nvread -x"
+fi
+
+# Do we have working TPM2.0?
+$pcrread >/dev/null 2>&1
+if [ $? -ne 0 ]; then
+	echo "Command execution failed, is TPM2.0 available?"
```
Comment (macpijan) [inline tools/test_tpm_ek_chain.sh:49]: @krystian-hebel It fails this way even when /dev/tpm0 is there, but tpm2-tools are not installed. We can improve logic and/or message

File: tools/test_tpm_ek_chain.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
```
Comment (macpijan) [inline tools/test_tpm_ek_chain.sh]:
```suggestion
#!/usr/bin/env bash
```

Comment (macpijan) [reply tools/test_tpm_ek_chain.sh]: Otherwise it would not work on my NixOS :P

File: tools/test_tpm_ek_chain.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+WORKDIR="$(mktemp -d)"
+declare -i MAX_CERT_SIZE=0
+declare -i CHAIN_LEN=0
+declare -i CHAIN_SIZE=0
+
+function cleanup {
+	rm -rf $WORKDIR
+	exit
+}
+
```
Comment (macpijan) [inline tools/test_tpm_ek_chain.sh:12]: I know you describe it in the issue, but it does not cost too many lines of code to check for the tools we need and inform user if they are missing. We need to make it as smooth as possible experiecne if we want to hope for some input from community.

File: tools/test_tpm_ek_chain.sh [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+done
+
+echo
+echo "Chain length: $CHAIN_LEN"
+echo "Total chain size: $CHAIN_SIZE bytes"
+echo "Size of the biggest certificate: $MAX_CERT_SIZE bytes"
+
+echo
+echo "Verifying whole chain:"
+
```
Comment (macpijan) [inline tools/test_tpm_ek_chain.sh:93]: I wonder if there is some generic way of telling us the vendor/model (or at least make it more simple to guess) without revealing too much from user in this log. User might not know what TMP is installed.

File: tools/test_tpm_ek_chain.sh [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+ca_files=$(find $WORKDIR -name '*.pem' ! -name 0.pem)
+
+# Pipe EK cert to hide name given by mktemp and let openssl speak for itself
+cat $WORKDIR/0.pem | openssl verify -CAfile <(cat $ca_files)
+
+cleanup
```
Comment (macpijan) [inline tools/test_tpm_ek_chain.sh:99]: @krystian-hebel I know you do not like it, but in the issue you could also provide one-liner to make it as simple as possible to download and execute the script. Or at least a set of command to download and execute. Right now the link is hidden in the text and assumes I know how to download and execute script, and have enough determination to do it by myself. Copying commands is alwasy a bit faster.

Lowering the energy I need to put into providing this information, increases the chances that I will do it, and you will receive it.

File: tools/test_tpm_ek_chain.sh (hunk 4 above)
Comment (krystian-hebel) [reply tools/test_tpm_ek_chain.sh:93]: It should be reported by `tpm2_getcap properties-fixed`, but this doesn't work on older tpm2-tools, at least for TPM I was testing.

File: tools/test_tpm_ek_chain.sh (hunk 5 above)
Comment (krystian-hebel) [reply tools/test_tpm_ek_chain.sh:99]: Agree, but let's merge this first so I can use the final URL. I'll address rest of the comments first thing tomorrow.

File: tools/test_tpm_ek_chain.sh (hunk 1 above)
Comment (krystian-hebel) [reply tools/test_tpm_ek_chain.sh:49]: Added test for tools availability.

File: tools/test_tpm_ek_chain.sh (hunk 2 above)
Comment (krystian-hebel) [reply tools/test_tpm_ek_chain.sh]: Changed. All those exotic OSes :slightly_smiling_face:

File: tools/test_tpm_ek_chain.sh (hunk 3 above)
Comment (krystian-hebel) [reply tools/test_tpm_ek_chain.sh:12]: Added.

File: tools/test_tpm_ek_chain.sh (hunk 4 above)
Comment (krystian-hebel) [reply tools/test_tpm_ek_chain.sh:93]: Added info to the issue, but what can be read from the TPM has somewhat limited usefulness.

**Review by macpijan — APPROVED** [review]

**Comment by macpijan** [comment]
LGTM. You can merge if needed.

**Pull request merged**